src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets

```diff
@@ -261,6 +261,37 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <!--
     ============================================================
+    ProjectReference Targets: Include transitive project references before 
+                              ResolveProjectReferences is called
+    - _ComputeTransitiveProjectReferences
+    - IncludeTransitiveProjectReferences
+    ============================================================
+    -->
+
+  <Target Name="_ComputeTransitiveProjectReferences"
+          DependsOnTargets="RunResolvePackageDependencies"
+          Returns="_TransitiveProjectReferences">
+    <ItemGroup>
+      <ProjectDefs Include="@(PackageDefinitions->WithMetadataValue('Type', 'project'))" />
+
+      <_TransitiveProjectReferences Include="%(ProjectDefs.Path)">
+        <ResolvedPath>%(ProjectDefs.ResolvedPath)</ResolvedPath>
+      </_TransitiveProjectReferences>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="IncludeTransitiveProjectReferences"
+          Condition="'$(ResolvePackageDependenciesForBuild)' == 'true' and '$(EnableResolvePackageDependencies)' == 'true' and Exists('$(ProjectAssetsFile)')"
+          DependsOnTargets="_ComputeTransitiveProjectReferences" 
+          BeforeTargets="AssignProjectConfiguration">
```

Comment:
`BeforeTargets="AssignProjectConfiguration"` ensures `ProjectReference` elements are created before they are processed into ResolvedProjectReferences. 

Alternative was to add this to the list of `ResolvePackageDependenciesForBuildDependsOn`, but this calls the target too late. However this means we have to copy the condition from `ResolvePackageDependenciesForBuild` to ensure we are not calling `RunResolvePackageDependencies` at inappropriate times. (E.g. Without this condition, the sdk build fails on building [xUnit.net.csproj](https://github.com/dotnet/sdk/blob/master/src/Dependencies/xUnit.net/xUnit.net.csproj)

Other alternative is to bypass ResolveProjectReferences and generate the `_ResolvedProjectReferencePaths` element directly, as is done in [Microsoft.NuGet.targets](https://github.com/NuGet/NuGet.BuildTasks/blob/dev/src/Microsoft.NuGet.Build.Tasks/Microsoft.NuGet.targets#L193). This currently does not work because of a bug calculated resolved paths (TODO insert bug).